src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -17,33 +17,74 @@ import { hasFetchedBoAndMoData, setFetchedBoMoData } from "../../utils/update/be
 
 import { CompanyPersonsWithSignificantControl } from "@companieshouse/api-sdk-node/dist/services/company-psc/types";
+import { BeneficialOwnerIndividual } from "../../model/beneficial.owner.individual.model";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
   try {
     logger.debugRequest(req, `${req.method} ${req.route.path}`);
 
     const appData: ApplicationData = getApplicationData(req.session);
     if (!hasFetchedBoAndMoData(appData)) {
-      await retrieveManagingOfficers(req, appData);
+      if (!appData.update) {
+        appData.update = {};
+      }
+      appData.update.review_beneficial_owners_individual = [];
+      appData.update.review_beneficial_owners_corporate = [];
+      appData.update.review_beneficial_owners_government_or_public_authority = [];
+
       await retrieveBeneficialOwners(req, appData);
+      await retrieveManagingOfficers(req, appData);
       setFetchedBoMoData(appData);
     }
+    if (appData.update?.review_beneficial_owners_individual?.length){
+      return checkAndReviewBeneficialOwner(appData, res);
+    }
 
     return res.render(config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE, {
       backLinkUrl: config.UPDATE_BENEFICIAL_OWNER_BO_MO_REVIEW_URL,
       templateName: config.UPDATE_BENEFICIAL_OWNER_TYPE_PAGE,
       ...appData,
-      noLists: true
+      noLists: true,
     });
   } catch (error) {
     logger.errorRequest(req, error);
     next(error);
   }
 };
 
+const checkBOIndividualValidation = (boi: BeneficialOwnerIndividual) => {
+  if (boi) {return true;}
+};
+
+export const checkAndReviewBeneficialOwner = (appData: ApplicationData, res: Response) => {
+  const beneficialOwnerReviewRedirectUrl = `${config.UPDATE_AN_OVERSEAS_ENTITY_URL
+    + config.UPDATE_REVIEW_BENEFICIAL_OWNER_INDIVIDUAL_PAGE
+    + config.REVIEW_BENEFICIAL_OWNER_INDEX_PARAM}`;
+
+  // Check last individual BO validates - in case back button is clicked
+  const boiLength = appData.beneficial_owners_individual?.length || 0;
+
+  const boiIndex: number = boiLength - 1;
+  if (appData.beneficial_owners_individual && (boiLength > 1) && !checkBOIndividualValidation(appData.beneficial_owners_individual[boiIndex])) {
+    return res.redirect(`${beneficialOwnerReviewRedirectUrl}${boiIndex}${config.REVIEW_BENEFICIAL_OWNER_REVIEW_PARAM}`);
+  }
+
+  // First review any retriewed individual bo:
+  if (boiLength >= 0){
+    const boi = appData.update?.review_beneficial_owners_individual?.pop();
+    let index = 0;
+
+    if (!appData.beneficial_owners_individual && boi) {
+      appData.beneficial_owners_individual = [boi];
+    } else if (appData.beneficial_owners_individual && boi) {
+      index = appData.beneficial_owners_individual.push(boi) - 1;
+    }
+    return res.redirect(`${beneficialOwnerReviewRedirectUrl}${index}${config.REVIEW_BENEFICIAL_OWNER_REVIEW_PARAM}`);
```

Comment:
would be wrong to redirect if really "boi" is falsy.

Reply:
"boi" will not be falsy in this method because "checkAndReviewBeneficialOwner" method will not if there is not "appData.update?.review_beneficial_owners_individual" on line 40